### ISSUE TITLE
Build breaks: error: no member named 'forced' in 'AVSubtitle'

### ISSUE THREAD

**Issue by yurivict** [MythTV/mythtv#638]
* **Platform**:
FreeBSD 13.1

* **MythTV version**:
32.0.60,1

* **Package version**:
n/a

* **Component**:
build

### What steps will reproduce the bug?
regular build

```
captions/subtitlereader.cpp:40:44: error: no member named 'forced' in 'AVSubtitle'
    if (!m_avSubtitlesEnabled && !subtitle.forced)
                                  ~~~~~~~~ ^
captions/subtitlereader.cpp:46:43: error: no member named 'forced' in 'AVSubtitle'
    if (!m_avSubtitlesEnabled && subtitle.forced)
                                 ~~~~~~~~ ^
2 errors generated.
```

### How often does it reproduce? Is there a required condition?
always

### What is the expected behaviour?
n/a

### Additional information
Log URL: http://beefy16.nyi.freebsd.org/data/131amd64-default/c1b60f972738/logs/errors/mythtv-32.0.60,1.log (IPv6 URL)

**Comment by linuxdude42** [comment]
I can't read your log file, but my guess is that you're finding the avcodec.h file from the system install, not the one from the version of FFmpeg built and used by MythTV.

**Comment by yurivict** [comment]
ffmpeg-4.4.2_8 is definitely installed.

Why are you bundling ffmpeg?

**Comment by linuxdude42** [comment]
You should still be able to build with a system FFmpeg installed. My FreeBSD 13.1 builder has ffmpeg-4.4.2_5,1 and it compiles without any problem. IIRC, the configure script tries to ensure that /usr/local comes last in the header include paths so that the bundled FFmpeg headers should be found before the system FFmpeg headers.

**Comment by yurivict** [comment]
MythTV version 32.0.60,1 is breaking in the FreeBSD package builders - this is why I created this ticket.
Somehow the system ffmpeg header gets included.

**Comment by benjsc** [comment]
Came across this one today after doing a system upgrade.
I've been able to reproduce this error:

`
gmake[2]: Entering directory '/mythtv/work/mythtv/mythtv-build/mythtv/libs/libmythtv
`
`
ccache clang++ -c -pipe -D_FILE_OFFSET_BITS=64 -DPIC -std=c++17 -faligned-new -DNDEBUG -O2 -fomit-frame-pointer -fPIC -DQT_DISABLE_DEPRECATED_BEFORE=0x050900 -msse -pthread -I/usr/local/include/soundtouch -g -Wall -Wextra -Wpointer-arith -Wno-constant-logical-operand -Wno-unused-value -Qunused-arguments -Wimplicit-fallthrough -faligned-allocation -I/usr/local/include/freetype2 -I/usr/local/include/libpng16 -I/usr/local/include -std=gnu++1z -pthread -fPIC -DMMX -Dusing_libcec -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -DHAVE_LIBBLURAY -DUSING_LIBCRYPTO -DUSING_LIBASS -DUSING_V4L2PRIME -DUSING_VAAPI -DUSING_OPENGL -DUSING_EGL -DUSING_MHEG -DUSING_FRONTEND -DUSING_V4L2 -DUSING_FFMPEG_THREADS -DUSING_OSS -DUSING_IPTV -DUSING_HDHOMERUN -DHDHOMERUN_HEADERFILE=\"libhdhomerun/hdhomerun.h\" -DHDHOMERUN_V2 -DUSING_SATIP -DUSING_VBOX -DUSING_CETON -DUSING_DVB -DUSING_BACKEND -DMTV_API -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_SQL_LIB -DQT_CORE_LIB -I. -I/usr/local/include/qt5 -I/usr/local/include/libxml2 -I.. -I../.. -I../.. -I../../external/FFmpeg -I. -I../libmyth -I../libmyth/audio -I../libmythbase -Impeg -Ichannelscan -Imheg -Idecoders -Iopengl -Iio -Icaptions -Ivisualisations -Ivisualisations/opengl -Ivisualisations/vulkan -Ivulkan -Idrm -Ioverlays -Irecorders -Irecorders/dvbdev -Irecorders/rtp -Irecorders/vbitext -Irecorders/HLS -I../libmythbase -I../libmythui -I../libmythupnp -I../libmythservicecontracts -I../../external/libmythdvdnav/dvdnav -I../../external/libmythdvdnav/dvdread -I../../external/libmythdvdnav/dvdnav -I../../external/libmythdvdnav/dvdread -I/usr/local/include/qt5/QtWidgets -I/usr/local/include/qt5/QtGui -I/usr/local/include/qt5/QtNetwork -I/usr/local/include/qt5/QtXml -I/usr/local/include/qt5/QtSql -I/usr/local/include/qt5/QtCore -Imoc -I/usr/local/include -I/usr/local/include -I/usr/local/lib/qt5/mkspecs/freebsd-clang -o obj/subtitlereader.o captions/subtitlereader.cpp
`
<pre>
captions/subtitlereader.cpp:40:44: error: no member named 'forced' in 'AVSubtitle'
    if (!m_avSubtitlesEnabled && !subtitle.forced)
                                  ~~~~~~~~ ^
captions/subtitlereader.cpp:46:43: error: no member named 'forced' in 'AVSubtitle'
    if (!m_avSubtitlesEnabled && subtitle.forced)
                                 ~~~~~~~~ ^
</pre>

This is with configure options:
`
   ./configure --prefix=/mythtv/env --cc=clang --cxx=clang++ --dvb-path=/usr/local/include --libxml2-path=/usr/local/include/libxml2 --enable-v4l2 --disable-audio-alsa --disable-indev=alsa --disable-outdev=alsa --extra-cflags=-I/usr/local/include --extra-ldflags=-L/usr/local/lib --compile-type=release --disable-vdpau --python=/usr/local/bin/python3.9 --enable-libmp3lame --enable-libx264
`

The issue is that :
`
-I /usr/local/include
`

Is included prior to:
`
-I../../external/FFmpeg
`

ie: 
`... -I/usr/local/include/libpng16 -I/usr/local/include ....  -I../../external/FFmpeg `

So the system version of ffmpeg is picked up first rather than the mythtv version.
This is on the fixes/32 branch rev: 44f88ed46834d1a5180c3bd9c1bda95c744b281d

Looking, it appears that _/usr/local/include_ is being pulled into _CXXFLAGS_ 
In this case. I hence trimmed the configure options to remove _-extra-cflags=-I/usr/local/include_ , still wouldn't build.
Played around to get a minimal amount of flags for my build but still it would fail with the same errors.

Looking at the [mythtv](https://www.freshports.org/multimedia/mythtv/) port/pkg, they have this  [patch](https://bugs.freebsd.org/bugzilla/attachment.cgi?id=236899&action=diff) from a [bug report](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=266671):

<pre>
--- configure  2022-09-02 15:59:37.029607000 +0100
+++ work/mythtv-32.0-60-ge5c974e402/mythtv/configure   2022-09-02 16:15:16.034232000 +0100
@@ -7786,7 +7786,7 @@
 fi

 if enabled freetype2; then
-    echo "FREETYPE_CFLAGS=$($pkg_config --cflags $pkg_config_flags freetype2)" >> $TMPMAK
+    echo "FREETYPE_CFLAGS=$($pkg_config --cflags $pkg_config_flags freetype2 |awk '{print $1, $2}')" >> $TMPMAK
     echo "FREETYPE_LIBS=$($pkg_config --libs $pkg_config_flags freetype2)" >> $TMPMAK
 fi

</pre>

Looking at the output of `pkg-config --cflags freetype2` gives:
<pre>
# pkg-config --cflags freetype2
-I/usr/local/include/freetype2 -I/usr/local/include/libpng16 -I/usr/local/include
</pre>

So this is where the _/usr/local/include_ is being added and the freebsd pkg strips it using awk.
I can confirm after trying a build with the following config options and the patch
`
 ./configure --prefix=/mythtv/env   --cc=clang --cxx=clang++ --dvb-path=/usr/local/include --libxml2-path=/usr/local/include/libxml2 --enable-v4l2 --disable-audio-alsa --compile-type=release --disable-vdpau --python=/usr/local/bin/python3.9
`

mythtv successfully builds.
So downstream have fixed the problem but it's still a problem in the mythtv repo. 
The ports fix, whilst it works for FreeBSD, may break other distros.
